### PR TITLE
few improvements + separation of concerns

### PR DESCRIPTION
**Pull request by bartes**

### CODE REVIEW

File: lib/castle/api.rb [hunk 1]
```diff
@@ -24,71 +32,73 @@ def initialize(cookie_id, ip, headers)
...
-        lang_version: lang_version,
+        lang_version: version,
         platform: RUBY_PLATFORM,
         publisher: 'castle',
         uname: @uname
       }
     end
 
-    def get_uname
+    def fetch_uname
```
Comment (nijikon) [inline lib/castle/api.rb]: `fetch_uname` -> `uname`

File: lib/castle/api.rb [hunk 2]
```diff
@@ -24,71 +32,73 @@ def initialize(cookie_id, ip, headers)
...
-          raise Castle::ApiError, 'Invalid response from Castle API'
-        end
-      end
     end
 
     def client_user_agent
-      @uname ||= get_uname
-      lang_version = "#{RUBY_VERSION} p#{RUBY_PATCHLEVEL} (#{RUBY_RELEASE_DATE})"
+      @uname ||= fetch_uname
+      version = "#{RUBY_VERSION} p#{RUBY_PATCHLEVEL} (#{RUBY_RELEASE_DATE})"
```
Comment (nijikon) [inline lib/castle/api.rb]: Move this to `ruby_version` method.

File: lib/castle.rb [hunk 3]
```diff
@@ -9,6 +9,7 @@
 require 'castle/client'
 require 'castle/errors'
 require 'castle/api'
+require 'castle/uname'
```
Comment (nijikon) [inline lib/castle.rb]: Rename this to `castle/system` move `uname` there and `ruby_version`

File: spec/lib/castle/client_spec.rb [hunk 4]
```diff
@@ -17,15 +17,21 @@ def delegate?
...
-  let(:client) { Castle::Client.new(request, nil) }
+  let(:client) { described_class.new(request, nil) }
   let(:review_id) { '12356789' }
 
-  it 'parses the request' do
-    expect(Castle::API).to receive(:new).with(cookie_id, ip,
-                                              "{\"X-Forwarded-For\":\"#{ip}\"}").and_call_original
+  describe 'parses the request' do
+    let(:api_data) { [cookie_id, ip, "{\"X-Forwarded-For\":\"#{ip}\"}"] }
+    let(:client) { described_class.new(request, nil) }
```
Comment (wallin) [inline spec/lib/castle/client_spec.rb]: No need to redefine `:client` here?

File: lib/castle/support/cookie_store.rb [hunk 5]
```diff
@@ -15,9 +15,10 @@ def [](key)
       def []=(key, value)
         @request.cookies[key] = value
         if value
-          @response.set_cookie(key, value: value,
-                                    expires: Time.now + (20 * 365 * 24 * 60 * 60),
-                                    path: '/')
+          @response.set_cookie(key,
+                               value: value,
+                               expires: time.now + (20 * 365 * 24 * 60 * 60),
```
Comment (wallin) [inline lib/castle/support/cookie_store.rb]: I guess we can put `(20 * 365 * 24 * 60 * 60)` in a constant

File: lib/castle/support/cookie_store.rb [hunk 6]
```diff
@@ -4,6 +4,7 @@ module Castle
   module CookieStore
     class Base
       def initialize(request, response)
+        @expire_time = 20 * 365 * 24 * 60 * 60
```
Comment (nijikon) [inline lib/castle/support/cookie_store.rb]: Make it a constant not variable, and please add a comment.

**Comment by bartes** [comment]
@nijikon ready for final review

**Pull request merged**